src/code/z_skelanime.c

```diff
@@ -1,176 +1,1680 @@
+        SkelAnime_DrawLimbSV(globalCtx, limbEntry->firstChildIndex, limbTable, actorDrawTable, updateMtxFunc,
+                             appendDlistFunc, actor, limbMatricies);
+    }
+
+    Matrix_Pull();
+
+    if (limbEntry->nextLimbIndex != LIMB_DONE) {
+        SkelAnime_DrawLimbSV(globalCtx, limbEntry->nextLimbIndex, limbTable, actorDrawTable, updateMtxFunc,
+                             appendDlistFunc, actor, limbMatricies);
+    }
+
+    func_800C6B54(sp34, globalCtx->state.gfxCtx, "../z_skelanime.c", 0x4F1);
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/SkelAnime_DrawLimbSV.s")
+#endif
+
+#ifdef NON_MATCHING
+void SkelAnime_DrawSV(GlobalContext* globalCtx, u32* limbTable, Vec3s* actorDrawTable, u32 dListCount,
+                      SkelAnime_LimbUpdateMatrix updateMtxFunc, SkelAnime_LimbAppendDlist appendDlistFunc,
+                      Actor* actor) {
+    SkelLimbEntry* rootLimb;
+    Gfx* skelDlist[2];
+    Vec3f pos;
+    Vec3s rot;
+    GraphicsContext* gfxCtx;
+    Mtx* limbMatricies;
+    Gfx* gfx[5];
+
+    limbMatricies = Graph_Alloc(globalCtx->state.gfxCtx, dListCount << 6);
+
+    if (limbTable == NULL) {
+        osSyncPrintf("\x1b[31m");
+        osSyncPrintf("Si2_draw_SV():skelがNULLです。\n"); // Si2_draw_SV (): skel is NULL.
+        osSyncPrintf("\x1b[m");
+        return;
+    }
+
+    gfxCtx = globalCtx->state.gfxCtx;
+
+    func_800C6AC4(gfx, globalCtx->state.gfxCtx, "../z_skelanime.c", 1294);
+
+    gSPSegment(gfxCtx->polyOpa.p++, 0xD, limbMatricies);
+    Matrix_Push();
+    rootLimb = SEGMENTED_TO_VIRTUAL(*limbTable);
+
+    pos.x = (f32)actorDrawTable[0].x;
+    pos.y = (f32)actorDrawTable[0].y;
+    pos.z = (f32)actorDrawTable[0].z;
+
+    rot = actorDrawTable[1];
+
+    skelDlist[0] = skelDlist[1] = rootLimb->dList;
+
+    if ((updateMtxFunc == NULL) || (updateMtxFunc(globalCtx, 1, &skelDlist[0], &pos, &rot, actor) == 0)) {
+        Matrix_TranslateThenRotateZYX(&pos, &rot);
+        if (skelDlist[0] != NULL) {
+            Matrix_ToMtx(limbMatricies, "../z_skelanime.c", 1327);
+            gSPMatrix(gfxCtx->polyOpa.p++, limbMatricies, G_MTX_LOAD);
+            gSPDisplayList(gfxCtx->polyOpa.p++, skelDlist[0]);
+            limbMatricies++;
+        } else {
+            if (skelDlist[1] != 0) {
+                Matrix_ToMtx(limbMatricies, "../z_skelanime.c", 1334);
+                limbMatricies++;
+            }
+        }
+    }
+
+    if (appendDlistFunc != 0) {
+        appendDlistFunc(globalCtx, 1, &skelDlist[1], &rot, actor);
+    }
+
+    if (rootLimb->firstChildIndex != LIMB_DONE) {
+        SkelAnime_DrawLimbSV(globalCtx, rootLimb->firstChildIndex, limbTable, actorDrawTable, updateMtxFunc,
+                             appendDlistFunc, actor, &limbMatricies);
+    }
+
+    Matrix_Pull();
+    func_800C6B54(gfx, globalCtx->state.gfxCtx, "../z_skelanime.c", 1347);
+}
+#else
+#pragma GLOBAL_ASM("asm/non_matchings/code/z_skelanime/SkelAnime_DrawSV.s")
+#endif
+#undef NON_MATCHING
```

Comment:
undef NON_MATCHING?

Reply:
498

Reply:
Yeah I forgot to remove these, will remove them in a few.

Reply:
Fixed in f2c65d6